m2cgen/interpreters/ruby/code_generator.py

```diff
@@ -0,0 +1,37 @@
+import contextlib
+
+from m2cgen.interpreters.code_generator import ImperativeCodeGenerator
+from m2cgen.interpreters.code_generator import CodeTemplate as CT
+
+
+class RubyCodeGenerator(ImperativeCodeGenerator):
+
+    tpl_var_declaration = CT("")
+    tpl_num_value = CT("${value}")
+    tpl_infix_expression = CT("(${left}) ${op} (${right})")
+    tpl_return_statement = tpl_num_value
+    tpl_array_index_access = CT("${array_name}[${index}]")
+    tpl_if_statement = CT("if ${if_def}")
+    tpl_else_statement = CT("else")
+    tpl_block_termination = CT("end")
+    tpl_var_assignment = CT("${var_name} = ${value}")
+
+    def add_function_def(self, name, args):
+        func_def = "def " + name + "("
+        func_def += ", ".join(args)
+        func_def += ")"
+        self.add_code_line(func_def)
+        self.increase_indent()
+
+    @contextlib.contextmanager
+    def function_definition(self, name, args):
+        self.add_function_def(name, args)
+        yield
+        self.add_block_termination()
+
+    def method_invocation(self, method_name, obj, args):
+        return ("(" + str(obj) + ")." + method_name +
```

Comment:
Why do we need these parentheses (the ones in code not in string)? It makes the code a bit harder to digest

Reply:
Btw, while I'm on it, why do we need parentheses around the object?

Reply:
It was my initial mistake to not use these parentheses 😃 . It was cough by one of our e2e tests. Then I realized that
`1.0 + 2.0.fdiv(3)` is not the same as `(1.0 + 2.0).fdiv(3)`.

Reply:
ah, makes perfect sense 👍